### PR TITLE
Ignoring power supplies FIX #753

### LINKED ISSUE #753: [Feature] Xbox controller's battery makes it switch to battery profile
### Have you tried?

- [Searching through existing/closed issues](https://github.com/AdnanHodzic/auto-cpufreq/issues) to see if your bug has already been already submitted? ✅
- If installation failed with [auto-cpufreq-installer](https://github.com/AdnanHodzic/auto-cpufreq/#auto-cpufreq-installer),have you tried installing auto-cpufreq using [Snap package](https://github.com/AdnanHodzic/auto-cpufreq/#snap-store)? ❌
- Have you tried [configuring auto-cpufreq](https://github.com/AdnanHodzic/auto-cpufreq/#configuring-auto-cpufreq)? ✅
- Have you tried suggestions in [troubleshooting section](https://github.com/AdnanHodzic/auto-cpufreq#troubleshooting)? ✅

### Error output:
```text
Add/paste error output in case of failed installation or other failing component
```
---

### System information:

Add/paste output of:

```
auto-cpufreq --debug

Using settings defined in /etc/auto-cpufreq.conf file

-------------------------------------------------------------------------------

Linux distro: Garuda Linux Soaring BirdOfPrey
Linux kernel: 6.10.0-zen1-2-zen
Processor: Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
Cores: 12
Architecture: x86_64
Driver: intel_pstate

------------------------------ Current CPU stats ------------------------------

CPU max frequency: 3700 MHz
CPU min frequency: 800 MHz

Core	Usage	Temperature	Frequency
CPU0      8.2%        37 °C       800 MHz
CPU1     14.9%        37 °C       800 MHz
CPU2      3.0%        37 °C       800 MHz
CPU3     14.0%        37 °C       800 MHz
CPU4      6.0%        38 °C       800 MHz
CPU5      9.3%        36 °C       800 MHz
CPU6     10.0%        37 °C       800 MHz
CPU7      6.0%        37 °C       800 MHz
CPU8      4.0%        37 °C       800 MHz
CPU9      8.9%        37 °C       800 MHz
CPU10      5.1%        38 °C       801 MHz
CPU11      7.9%        36 °C       800 MHz

CPU fan speed: 91 RPM

CPU fan speed: 0 RPM

auto-cpufreq version: Version         : 2.3.0-1

Python: 3.12.4
psutil package: 6.0.0
platform package: 1.0.8
click package: 8.1.7
distro package: 1.9.0

Computer type: Desktop
Battery is: discharging

auto-cpufreq system resource consumption:
cpu usage: 0.0 %
memory use: 0.12 %

Total CPU usage: 2.0 %
Total system load: 1.55
Average temp. of all cores: 44.17 °C

Currently using: powersave governor
Currently turbo boost is: off

-------------------------------------------------------------------------------
```

Also please be descriptive about the issue you're reporting, i.e: what you tried & what's the expected behaviour.

---

Few months ago I have copied default config from readme to my conf file on my desktop pc and only changed turboboost to always. I was playing games in Windows vm with xbox wireless adapter passed to it back then. 

Last week I tried gaming natively on Linux and installed xone driver (xone-dkms in aur) which reports battery level of controller. I then tried playing FS22 (pretty CPU intensive) and it worked great (90-130 fps)... until I connected controller and fps dropped to 30. Then I tried connecting it with wire and it worked normally (because controller doesn't report battery usage when plugged). I noticed that cpu frequency was a bit low (800 MHz). Then I went to look at output of `auto-cpufreq status` and saw that it is using powersave governor and that (controller's) battery was discharging.

In debug output I noticed that program detects computer type. Why doesn't it then just ignore the battery entry in config if desktop type is detected?

I tried connecting my motorcycle helmet intercom to pc and its battery wasn't detected in auto-cpufreq.

Then I looked into `/sys/class/power_supply` and found that xone has made an entry named `gip0.0`
Contents of uevent to help you filter it out:
```
DEVTYPE=power_supply
POWER_SUPPLY_NAME=gip0.0
POWER_SUPPLY_TYPE=Battery
POWER_SUPPLY_STATUS=Discharging
POWER_SUPPLY_CAPACITY_LEVEL=High
POWER_SUPPLY_SCOPE=Device
POWER_SUPPLY_MODEL_NAME=Microsoft Xbox Controller
```

### PR DESCRIPTION
**Pull request by PurpleWazard**
this PR add an option in the config file to ignore specific power supplies in /sys/class/power_supply. this fixes issue #753 

i also updated the example config file and updated the read me.

### CODE REVIEW

File: auto-cpufreq.conf-example [hunk 1]
```diff
@@ -22,6 +22,17 @@ energy_performance_preference = performance
 # turbo boost setting. possible values: always, auto, never
 turbo = auto
 
+
+# this is for ignoring controlers and other connected devices battery from effecting
```
Comment (timrosu) [inline auto-cpufreq.conf-example:26]: spelling correction: controllers and affecting

Comment (PurpleWazard) [reply auto-cpufreq.conf-example:26]: Thx!

**Comment by AdnanHodzic** [comment]
@PurpleWazard can you please resolve conflicts so we can get this merged? Thanks!

File: auto-cpufreq.conf-example (hunk 1 above)
Comment (timrosu) [reply auto-cpufreq.conf-example:26]: np

Comment (PurpleWazard) [reply auto-cpufreq.conf-example]: fixed!

**Comment by PurpleWazard** [comment]
@AdnanHodzic merge conflict is resolved

File: 
Comment (shadeyg56) [inline :1]: Could this weird file be removed? It was added in 80bb137

Comment (PurpleWazard) [reply :1]: yep just did. im unsure how it came to be.

**Review by shadeyg56 — APPROVED** [review]

**Comment by shadeyg56** [comment]
LGTM. Merging now since it looks like Adnan has already looked at this.

Thanks for the PR! You'll be credited in the next release as usual.

**Pull request merged**